tests/unit/externalRequestsTest.js

```diff
@@ -1,61 +1,274 @@
+        const ctx = {
+          status: 200,
+          state: {
+            metaData: {
+              requests: null
+            }
+          }
+        }
+        await orchestrateMappingResult(ctx)
+
+        t.equals(ctx.orchestrations)
+        t.end()
+      }
+    )
+
+    t.test(
+      'should not do any orchestrations when the requests object does not have the response requests',
+      async t => {
+        const ctx = {
+          status: 200,
+          state: {
+            metaData: {
+              requests: {
+                response: []
+              }
+            }
+          },
+          request: {}
+        }
+
+        await orchestrateMappingResult(ctx)
+
+        t.equals(ctx.orchestrations, undefined)
+        t.end()
+      }
+    )
+
+    t.test(
+      'should not do any orchestrations when request url/method/id is not supplied',
+      async t => {
+        const ctx = {
+          status: 200,
+          state: {
+            metaData: {
+              requests: {
+                response: [{}]
+              }
+            }
+          },
+          request: {}
+        }
+        await orchestrateMappingResult(ctx)
+
+        t.equals(ctx.orchestrations.length, 0)
+        t.end()
+      }
+    )
+
+    t.test(
+      'should send requests and record the orchestrations (to a server that is down) (statusText should be Failed)',
+      async t => {
+        const url = 'http://localhost:8000/'
+        const url2 = 'http://localhost:9000/'
+        const method = 'PUT'
+        const id = '1233243'
+
+        const ctx = {
+          status: 200,
+          state: {
+            metaData: {
+              requests: {
+                response: [
+                  {
+                    config: {
+                      url: url,
+                      method: method
+                    },
+                    id: id,
+                    primary: true
+                  },
+                  {
+                    config: {
+                      url: url2,
+                      method: method
+                    },
+                    id: id
+                  }
+                ]
+              }
+            }
+          },
+          request: {
+            header: {
+              'x-openhim-transactionid': '1232244'
+            }
+          },
+          response: {
+            headers: {}
+          },
+          body: {},
+          set: (key, value) => {
+            ctx.response.headers[key] = value
+          }
+        }
+
+        await orchestrateMappingResult(ctx)
+
+        t.equals(ctx.orchestrations.length, 2)
+        t.match(ctx.orchestrations[0].error.message, /ECONNREFUSED/)
+        t.match(ctx.body, /Failed/)
+        t.end()
+      }
+    )
+
+    t.test(
+      'should send requests and record the orchestrations (response statusText - Successful)',
+      async t => {
+        const url = 'http://localhost:8000/'
+        const method = 'PUT'
+        const id = 'Patient'
```

Comment:
I dont think its needed to declare this as a variable if its only being used once

Reply:
done in https://github.com/jembi/openhim-mediator-mapping/pull/39/commits/f6e854686999d065f57f20f9f1af3f9e1318f21e